application/models/services/UserManager.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace LimeSurvey\Models\Services;
+
+use LSWebUser;
+use Permission;
+use User;
+
+/**
+ * Service class for managing users and their permissions
+ */
+class UserManager
+{
+    /** @var LSWebUser the user managing other users */
+    private $managingUser;
+
+    /** @var User the user being handled */
+    private $targetUser;
+
+    /**
+     * @param LSWebUser $managingUser
+     * @param User|null $targetUser
+     */
+    public function __construct(
+        LSWebUser $managingUser,
+        User $targetUser = null
+    ) {
+        $this->managingUser = $managingUser;
```

Comment:
managingUser must be current user no  ?

Reply:
Yes, that's when initialized... 
We could automate it a bit more to have the managinsuer to be set to current user by default